antlr-kotlin-runtime/src/jsMain/kotlin/com/strumenta/antlrkotlin/runtime/BitSet.kt

```diff
@@ -1,85 +1,108 @@
 // Copyright 2017-present Strumenta and contributors, licensed under Apache 2.0.
 // Copyright 2024-present Strumenta and contributors, licensed under BSD 3-Clause.
-
 package com.strumenta.antlrkotlin.runtime
 
 import js.core.delete
 import org.antlr.v4.kotlinruntime.misc.MurmurHash
 
-public actual class BitSet {
-  private val wrapped = js("[]").unsafeCast<Array<Boolean>>()
+@Suppress("SimplifyBooleanWithConstants")
+public actual class BitSet actual constructor(size: Int) {
```

Comment:
Surprisingly a more compact bitset implementation with `Uint8Array` doesn't give any performance boost until we get to 100k+ bits. On the contrary, this dynamic array version is a lot faster with 1-5000 bits.

Reply:
Would it make sense to look at the size and based on that go for one or the other or would that be overkill?

Reply:
It's overkill. You'll never have a bit set that large.